### PR TITLE
fix(virtio-mem):interval intersection in slots_intersecting_range

### PR DESCRIPTION
**Pull request by kalyazin**
The previous implementation checked whether either slot endpoint fell inside the requested range. This missed the containment case where a slot fully contains the range (neither endpoint inside it), causing update_kvm_slots to silently skip KVM slot registration/removal for any block not aligned to a slot boundary.

Replace the two addr_in_range endpoint checks with a proper half-open interval intersection test: slot_start < range_end && range_start < slot_end.

Remove the now-unused addr_in_range helper and add a table-driven unit test covering boundary, interior, cross-slot, full-region, outside, and zero-length ranges.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] I have read and understand [CONTRIBUTING.md][3].
- [ ] I have run `tools/devtool checkbuild --all` to verify that the PR passes
  build checks on all supported architectures.
- [ ] I have run `tools/devtool checkstyle` to verify that the PR passes the
  automated style checks.
- [ ] I have described what is done in these changes, why they are needed, and
  how they are solving the problem in a clear and encompassing way.
- [ ] I have updated any relevant documentation (both in code and in the docs)
  in the PR.
- [ ] I have mentioned all user-facing changes in `CHANGELOG.md`.
- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] When making API changes, I have followed the
  [Runbook for Firecracker API changes][2].
- [ ] I have tested all new and changed functionalities in unit tests and/or
  integration tests.
- [ ] I have linked an issue to every new `TODO`.

______________________________________________________________________

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,11 @@ and this project adheres to
 - [#5780](https://github.com/firecracker-microvm/firecracker/pull/5780): Fixed
   missing `/sys/devices/system/cpu/cpu*/cache/*` in aarch64 guests when running
   on host kernels >= 6.3 with guest kernels >= 6.1.156.
+- [#xxxx](https://github.com/firecracker-microvm/firecracker/pull/xxxx): Fixed
```
Comment (Manciukic) [inline CHANGELOG.md]:
```suggestion
- [#5793](https://github.com/firecracker-microvm/firecracker/pull/xxxx): Fixed
```

Comment (kalyazin) [reply CHANGELOG.md]: Apparently I pushed a different branch instead of this! Fixed.

**Comment by kalyazin** [comment]
> I'm wondering if we could have caught this bug from `test_virtio_mem_configs`. Maybe moving `check_memory_usable` inside the loop? or maybe the guest kernel always behaves sequentially so that the bug is never triggered.

Just tried that with the fix reverted, but it doesn't seem to detect.

**Comment by kalyazin** [comment]
> > Just tried that with the fix reverted, but it doesn't seem to detect.
> 
> if we could add an integ test to trigger it, that would be best. I'm not sure how though. It'll probably need some kind of "fuzzing" to trigger the condition, but I would time-box it.

Yeah, such a test would be quite involved. We should discover somehow the GPA of the unplugged memory and access it via /dev/mem and we should be able to detect that KVM/Firecracker did the right thing of not giving access to the page (crash?) somehow. I'm not sure the level of complexity warrants the coverage we'd get. Also the test would be very specialised for this particular scenario.
